collector/providers/odigosfileprovider/provider.go

```diff
@@ -0,0 +1,138 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// forked from go.opentelemetry.io/collector/confmap/provider/fileprovider
+package odigosfileprovider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/zap"
+)
+
+const schemeName = "file"
+
+type provider struct {
+	wg sync.WaitGroup
```

Comment:
nit: since we are aiming for one goroutine, this can be defined as:
```suggestion
	done chan struct{}
```

Reply:
good call